src/controllers/sourceController.ts

```
@@ -17,5 +17,37 @@ const router = express.Router();
 //Endpoints
 router.get('/api/sources', getAllSources);
 
+router.post('/api/sources/update-active', async (req: Request, res: Response) => {
+    
+    // Extract `id` and `active` from the request body
+    const { id, active } = req.body;
+
+    // Validate `id`
+    if (typeof id !== 'number' || isNaN(id)) {
```

Here you should throw an error and send it as error response in the catch block